src/test/java/com/gooddata/AbstractGoodDataAT.java

```diff
@@ -30,20 +27,22 @@
             "sdktest " + new LocalDate() + " " + System.getenv("BUILD_NUMBER");
 
     protected static final GoodData gd =
-            new GoodData(getProperty("host"), getProperty("login"), getProperty("pass")) {
-                /**
-                 * had to be overriden to access connectionManager, to test how many connections were not closed
-                 */
-                @Override
-                protected HttpClient createHttpClient(String login, String password, String hostname, int port, String protocol, HttpClientBuilder builder) {
-                    PoolingHttpClientConnectionManager httpClientConnectionManager = new PoolingHttpClientConnectionManager();
-                    final HttpHost host = new HttpHost(hostname, port, protocol);
-                    final HttpClient httpClient = builder.setConnectionManager(httpClientConnectionManager).build();
-                    final SSTRetrievalStrategy strategy = new LoginSSTRetrievalStrategy(httpClient, host, login, password);
-                    connManager = httpClientConnectionManager;
-                    return new GoodDataHttpClient(httpClient, strategy);
-                }
-            };
+            new GoodData(
+                    new GoodDataEndpoint(getProperty("host")),
+                    new LoginPasswordAuthentication(getProperty("login"), getProperty("pass")) {
+                        /**
+                         * had to be overriden to access connectionManager, to test how many connections were not closed
+                         */
+                        @Override
+                        public HttpClient createHttpClient(final GoodDataEndpoint endpoint, final HttpClientBuilder builder) {
+                            PoolingHttpClientConnectionManager httpClientConnectionManager = new PoolingHttpClientConnectionManager();
+                            final HttpClientBuilder builderWithManager = builder.setConnectionManager(httpClientConnectionManager);
+                            connManager = httpClientConnectionManager;
+
+                            return super.createHttpClient(endpoint, builderWithManager);
+                        }
+                    }, new GoodDataSettings());
```

Comment:
Can you please create constructor GoodData(GoodDataEndpoint, Authentication) and use it here?